aws-datastore/src/androidTest/java/com/amplifyframework/datastore/MultiAuthSyncEngineInstrumentationTest.java

```diff
@@ -0,0 +1,1036 @@
+            token.set(idToken.getTokenValue());
+        } catch (IOException exception) {
+            LOG.warn("An error occurred while trying to authenticate against OIDC provider", exception);
+        }
+    }
+
+    /**
+     * Test tear-down activities.
+     */
+    @AfterClass
+    public static void cleanupAfterAllTests() {
+        Log.i("TearDown", "Cleaning up databases");
+        for (String db : getApplicationContext().databaseList()) {
+            Log.i("TearDown", "Removing " + db);
+            getApplicationContext().deleteDatabase(db);
+            Log.i("TearDown", db + " removed");
+        }
+        Log.i("TearDown", "Cleanup completed.");
+    }
+
+    private void verifyScenario(Class<? extends Model> modelType,
+                                  boolean signInToCognito,
+                                  boolean signInWithOidc,
+                                  AuthorizationType expectedAuthType) throws AmplifyException, IOException {
+        configure(modelType, signInToCognito, signInWithOidc, expectedAuthType);
+        String modelId = UUID.randomUUID().toString();
+        Model testRecord = createRecord(modelType, modelId);
+        HubAccumulator expectedEventAccumulator = null;
+        if (expectedAuthType != null) {
+            expectedEventAccumulator =
+                HubAccumulator
+                    .create(HubChannel.DATASTORE, publicationOf(modelType.getSimpleName(), testRecord.getId()), 1)
+                    .start();
+        } else {
+            expectedEventAccumulator = HubAccumulator
+                    .create(HubChannel.DATASTORE, networkStatusFailure(), 1)
+                    .start();
+        }
+        dataStore.start();
+        dataStore.save(testRecord);
+        expectedEventAccumulator.await(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        assertFalse(requestInterceptor.hasUnexpectedRequests());
+        // Sign the user out if sign-in was required.
+        if (signInToCognito) {
+            auth.signOut(AuthSignOutOptions.builder().build());
+        }
+    }
+
+    /**
+     * Create a instance of the model using the private constructor via reflection.
+     * @return A instance of the model being tested.
+     */
+    private Model createRecord(Class<?> modelType, String modelId) {
+        try {
+            Constructor<?> constructor = modelType.getDeclaredConstructors()[0];
+            constructor.setAccessible(true);
+            String recordDetail = "IntegTest-" + modelType.getSimpleName() + " " + RandomString.string();
+            // Most constructor for the multiauth test models have 2 parameters.
+            if (constructor.getParameterCount() == 2) {
+                return (Model) constructor.newInstance(modelId, recordDetail);
+            }
+            // For the one exception, we just pass a null for the extra parameter.
+            return (Model) constructor.newInstance(modelId, recordDetail, null);
```

Comment:
This doesn't quite sit right with me.
Assuming that they all have the format of
```
model {
  id: ID!
  name: String!
}
```

then perhaps you can do something along the line of:
```
Map<String, String> model = new HashMap<>();
model.put("id", id);
model.put("name", name);
String serialized = gson.toJson(model);
return gson.fromJson(serialized, modelType);
```

Reply:
that's a good idea. That works nicely and I can remove all of the reflection hack.